build/bin/sage-get-system-packages

```diff
@@ -10,6 +11,12 @@ SPKGS="$*"
 if [ -z "$SAGE_ROOT" ]; then
     SAGE_ROOT=`pwd`
 fi
+
+# We source sage-build-env-config because we need to know if
+# --enable-system-site-packages was passed to ./configure.
+# (If not, we want to hide those packages.)
+. "${SAGE_ROOT}/build/bin/sage-build-env-config"
```

Comment:
That's not good; this script must also work without error in an unconfigured source tree!

Reply:
Better to just assume that `ENABLE_SYSTEM_SITE_PACKAGES` is already set in the environment.

Reply:
I changed this but I'm not sure it makes complete sense. For example `write-dockerfile.sh` now always omits the python packages unless you pass it `ENABLE_SYSTEM_SITE_PACKAGES=yes` manually. That itself is not a big deal but who knows if some other issue is lurking.

Reply:
Thanks. Here's an update for tox.ini, which invokes the write-dockerfile.sh
```
--- a/tox.ini
+++ b/tox.ini
@@ -187,6 +187,8 @@ setenv =
     develop:                 EXTRA_SAGE_PACKAGES_4=_develop $(head -n 1 build/pkgs/_develop/dependencies)
     minimal:               SAGE_PACKAGE_LIST_ARGS=_prereq
     maximal:               SAGE_PACKAGE_LIST_ARGS=:standard: :optional:
+    sitepackages:              ENABLE_SYSTEM_SITE_PACKAGES=yes
+    sitepackages:                CONFIG_CONFIGURE_ARGS_SITEPACKAGES=--enable-system-site-packages
     conda-environment:     SAGE_PACKAGE_LIST_ARGS=_prereq
     # Whether to add the system packages needed for bootstrapping
                            EXTRA_SAGE_PACKAGES_0=_bootstrap
@@ -601,7 +603,7 @@ setenv =
     #
     # Resulting full configuration args, including EXTRA_CONFIGURE_ARGS from the user environment
     #
-                      CONFIGURE_ARGS=--enable-experimental-packages --enable-download-from-upstream-url {env:CONFIG_CONFIGURE_ARGS_ROOT:} {env:CONFIG_CONFIGURE_ARGS_1:} {env:CONFIG_CONFIGURE_ARGS_2:} {env:EXTRA_CONFIGURE_ARGS:}
+                      CONFIGURE_ARGS=--enable-experimental-packages --enable-download-from-upstream-url {env:CONFIG_CONFIGURE_ARGS_ROOT:} {env:CONFIG_CONFIGURE_ARGS_SITEPACKAGES:} {env:CONFIG_CONFIGURE_ARGS_1:} {env:CONFIG_CONFIGURE_ARGS_2:} {env:EXTRA_CONFIGURE_ARGS:}
     #
     # Resulting EXTRA_SAGE_PACKAGES
     #
```

Reply:
You should be able to push to my branch if you want to add that to the repo, dima was able to.

Reply:
OK, will do.

Reply:
done in e9c4710791